core/contractsapi/abstract_api.go

```diff
@@ -163,6 +163,10 @@ func (s *Stream) SetReadVisibility(ctx context.Context, visibility util.Visibili
 	return s.insertMetadata(ctx, types.ReadVisibilityKey, types.NewMetadataValue(int(visibility)))
 }
 
+func (s *Stream) InsertDefaultBaseDate(ctx context.Context, baseDate string) (transactions.TxHash, error) {
```

Comment:
```suggestion
func (s *Stream) SetDefaultBaseDate(ctx context.Context, baseDate string) (transactions.TxHash, error) {
```

Set is more appropriate as it is something being set and overridden if being set again.